src/main/java/at/hannibal2/skyhanni/features/misc/reminders/ReminderManager.kt

```diff
@@ -0,0 +1,256 @@
+            action(args.drop(2), storage[args.drop(1).first()]!!).apply {
+                listReminders(listPage)
+                sendMessage(this)
+            }
+        } else if (storage[args.first()] == null) {
+            return ChatUtils.userError("Reminder not found")
+        } else {
+            sendMessage(action(args.drop(1), storage[args.first()]!!))
+        }
+    }
+
+    private fun removeReminder(args: List<String>) = actionReminder(
+        args,
+        "remove",
+        "[id]",
+    ) { _, reminder ->
+        storage.values.remove(reminder)
+        "§cReminder deleted."
+    }
+
+    private fun editReminder(args: List<String>) = actionReminder(
+        args,
+        "edit",
+        "[id]",
+        "[reminder]",
+    ) { arguments, reminder ->
+        reminder.reason = arguments.joinToString(" ")
+        "§6Reminder edited."
+    }
+
+    private fun moveReminder(args: List<String>) = actionReminder(
+        args,
+        "move",
+        "[id]",
+        "[time]",
+    ) { arguments, reminder ->
+        val time = parseReminder(arguments.first()) ?: return@actionReminder "§cInvalid time format!"
+        reminder.remindAt = SimpleTimeMark.now().plus(time)
+        "§6Reminder moved to ${time.format()}"
+    }
+
+    private fun help() {
+        createDivider().send()
+        "§6SkyHanni Reminder Commands:".asComponent().send()
+        "§e/shremind <time> <reminder> - §bCreates a new reminder".asComponent().send()
+        "§e/shremind list <page> - §bLists all reminders".asComponent().send()
+        "§e/shremind remove <id> - §bRemoves a reminder".asComponent().send()
+        "§e/shremind edit <id> <reminder> - §bEdits a reminder".asComponent().send()
+        "§e/shremind move <id> <time> - §bMoves a reminder".asComponent().send()
+        "§e/shremind help - §bShows this help message".asComponent().send()
+        createDivider().send()
+    }
+
+    @SubscribeEvent
+    fun onSecondPassed(event: SecondPassedEvent) {
+        val remindersToSend = mutableListOf<IChatComponent>()
+
+        for ((id, reminder) in getSortedReminders()) {
+            if (!reminder.shouldRemind(config.interval.minutes)) break
+            reminder.lastReminder = SimpleTimeMark.now()
+            var actionsComponent: IChatComponent? = null
+
+            if (!config.autoDeleteReminders) {
+                actionsComponent = Text.join(
+                    " ",
+                    "§a✔".asComponent {
+                        hover = "§7Click to dismiss".asComponent()
+                        command = "/shremind remove $id"
+                    }.wrap("§8[", "§8]"),
+                    " ",
+                    "§e§l⟳".asComponent {
+                        hover = "§7Click to move".asComponent()
+                        suggest = "/shremind move $id 1m"
+                    }.wrap("§8[", "§8]"),
+                )
+            } else {
+                storage.remove(id)
```

Comment:
Are you sure this will not cause concurrent modification exceptions? editing a map while iterating over it. I have not encountered some while testing though

Reply:
Nevermind, just checked, `sortedWith()` creates a copy